libs/modelviewer/src/components/legend/Legend.tsx

```diff
@@ -82,27 +82,34 @@ const Style = {
 };
 
 export const Legend = () => {
-  const { tagList, toggleTags, filterTags, notFoundTagList } = useSelectionContext();
+  const { viewNodes, tagList, toggleTags, filterTags, notFoundTagList } =
+    useSelectionContext();
   const [isExpanded, setIsExpanded] = useState(true);
 
   const legendData = useMemo(() => {
     return tagList.reduce((acc, item) => {
       if (!item.status) return acc;
-
-      if (acc[item.status]) {
-        acc[item.status].tags.push(item.tagNo);
-      } else {
-        acc[item.status] = {
-          status: item.status,
-          tags: [item.tagNo],
-          color: item.color || '',
-        };
+      if (!notFoundTagList.includes(item)) {
+        if (acc[item.status]) {
```

Comment:
Just a code suggestion, but could this be a simpler way of accomplishing your goal? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/groupBy

Reply:
Seems like this is not supported by some versions of Safari and Samsung browser? 
This may be used on the iPads/Tablets/Computers that are in use by  the workers.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/groupBy